### DISCUSSION TITLE
Some new user questions about tutorial

### DISCUSSION THREAD

**Discussion by Weihe-BNL** [PandABlocks/PandABlocks-client#35] [Q&A]
Hello,

I am a new user of PandAblocks and just start going through the tutorial with Python 3.10 under windows terminal. Unfortunately, I got stuck at the beginning. There are some (stupid) questions:

1) in the 'Installation Tutorial', there is the commend 'pandablocks --version' to check the version. It seems not work for me, I used 'pandablocks.__version__' to go through. Is it due to my setting problem?

2) in the 'Commandline load/save Tutorial', the save command is listed as '$ pandablocks save <hostname> <save_file>'. However, my pandablock cannot recognize the 'save' commend

Can anyone give some suggestions?

Thanks,
Roy

**Comment by AlexanderWells-diamond** [comment]
For `1`, it sounds like you are inside a Python shell rather than a Bash shell? There is no command line tool of the name `pandablocks.version`. There's also no `version` attribute on the `pandablocks` module, so I'm not entirely sure what's happened. Could you please copy-paste your terminal output from trying to run the `--version` command?

For `2`, I'm not sure exactly where you are. The docs currently give the save command syntax as this:
```
$ pandablocks save <hostname> <save_file>
```

Could you please try that syntax, and if it doesn't work please copy-paste the output from your terminal into your reply.

**Reply by Weihe-BNL** [reply]
Hello Alexander:

Thank you very much for the answer. For my previous question 1, it was solved by running it under Bash shell. However, for question two, the system said the 'pandablocks' do not have a 'save' command. I was using version 0.2.1. Attached is my screenshot. Could you give some suggestions?
<img width="960" alt="screenshot" src="https://user-images.githubusercontent.com/112650236/198371689-8638932c-00f2-4d21-a464-f3d60b071a8c.png">

**Reply by AlexanderWells-diamond** [reply]
It looks like you had an error trying to install the module from Github (`errno=Resource temporarily unavailable`). This is confirmed by the version number still being that of the last proper release. The current version of the master branch is `pandablocks, version 0.2.1+107.g4e7a1ef`.

I would suggest first uninstalling the existing pandablocks and then trying to install from Git. Or just recreate the virtual environment from scratch.

As an aside, it appears there has never been a proper tagged release of the "save" command (nor several others). @coretl Any thoughts on whether we should tag+release?

**Reply by coretl** [reply]
Tag and release is a good idea. Do you mind doing this @AlexanderWells-diamond ?

**Reply by AlexanderWells-diamond** [reply]
Version 0.3.0 is now released to PyPi. @Roy-de-BNL you should now be able to install the latest version more easily.

**Reply by Weihe-BNL** [reply]
Hello Alexander:

I tried to run Version 0.3.0 and got attached error message. Have I missed some modules?

Thanks,
Weihe
<img width="604" alt="Untitled" src="https://user-images.githubusercontent.com/112650236/198689547-2cda9c31-aff6-484a-a80f-8e3696395173.png">

**Reply by Weihe-BNL** [reply]
Problem solved by installing the 'softioc' and 'hd5' modules

**Reply by AlexanderWells-diamond** [reply]
Apologies, version 3.0.0 had a packaging mistake in it. Version 3.0.1 should fix this.
